### PR TITLE
systemd: Tighten up sandboxing for eos-companion-app.serivce

### PR DESCRIPTION
**Pull request by smspillaz**
bwrap needs a few unfortunate capabilities such as CAP_SYS_ADMIN,
CAP_SETUID/CAP_SETGID, but we can apply most of the other sandboxing
options asides from the system call filter.

https://phabricator.endlessm.com/T23201

### CODE REVIEW

File: data/eos-companion-app.service [hunk 1]
```diff
@@ -6,7 +6,29 @@ ConditionPathExists=/etc/avahi/services/companion-app.service
 [Service]
 ExecStart=/usr/bin/dbus-run-session /usr/bin/flatpak run --no-desktop --no-a11y-bus --no-documents-portal com.endlessm.CompanionAppService
 User=companion-app-helper
+# This does not appear to cause any problems when writing
+# to /var/lib/eos-companion-app, remove if it does. We need
+# CAP_SETUID and CAP_SETGID as bwrap is setuid
```
Comment (pwithnall) [inline data/eos-companion-app.service]: Nitpick: Would be good to briefly motivate the inclusion of `CAP_NET_ADMIN`, `CAP_SYS_ADMIN` and `CAP_SYS_CHROOT` too.

File: data/eos-companion-app.service [hunk 2]
```diff
@@ -6,7 +6,29 @@ ConditionPathExists=/etc/avahi/services/companion-app.service
...
+CapabilityBoundingSet=CAP_NET_ADMIN CAP_SYS_ADMIN CAP_SYS_CHROOT CAP_SETUID CAP_SETGID
+MemoryDenyWriteExecute=yes
+NoNewPrivileges=yes
+PrivateDevices=yes
+# Breaks xapian-bridge
+#PrivateNetwork=yes
```
Comment (pwithnall) [inline data/eos-companion-app.service]: Nitpick: Maybe flip the sense of this, and instead do:
```
# xapian-bridge needs local HTTP networking
PrivateNetwork=no
```

File: data/eos-companion-app.service [hunk 3]
```diff
@@ -6,7 +6,29 @@ ConditionPathExists=/etc/avahi/services/companion-app.service
...
 PrivateTmp=yes
+PrivateUsers=yes
+ProtectControlGroups=yes
+ProtectHome=yes
+ProtectKernelModules=yes
+# Breaks bwrap
+#ProtectKernelTunables=yes
```
Comment (pwithnall) [inline data/eos-companion-app.service]: Nitpick: Similarly, I’d probably flip this one around too:
```
# bwrap mounts /proc
ProtectKernelTunables=no
```

File: data/eos-companion-app.service [hunk 4]
```diff
@@ -6,7 +6,29 @@ ConditionPathExists=/etc/avahi/services/companion-app.service
...
+ProtectSystem=yes
+# xapian-bridge needs AF_INET, we should probably
+# allow AF_INET6 too
+RestrictAddressFamilies=AF_UNIX AF_INET AF_INET6
```
Comment (pwithnall) [inline data/eos-companion-app.service]: In https://github.com/endlessm/eos-updater/commit/fc03bd7be598d0a3114cb56e1b40113fca2a8f2e, Rob said that bwrap needed `AF_NETLINK`. Is that not the case?

Comment (smspillaz) [reply data/eos-companion-app.service]: It appears not to. Perhaps that is only required when installing flatpaks.

File: data/eos-companion-app.service (hunk 2 above)
Comment (smspillaz) [reply data/eos-companion-app.service]: Good idea (also good to add more of an explanation in the comment)

File: data/eos-companion-app.service (hunk 1 above)
Comment (smspillaz) [reply data/eos-companion-app.service]: Done

File: data/eos-companion-app.service (hunk 3 above)
Comment (smspillaz) [reply data/eos-companion-app.service]: Ditto

File: data/eos-companion-app.service [hunk 5]
```diff
@@ -6,7 +6,30 @@ ConditionPathExists=/etc/avahi/services/companion-app.service
...
 User=companion-app-helper
+# This does not appear to cause any problems when writing
+# to /var/lib/eos-companion-app, remove if it does. We need
+# CAP_SETUID and CAP_SETGID as bwrap is setuid. CAP_SYS_ADMIN,
+# CAP_NET_ADMIN and CAP_SYS_CHROOT are also used by bwrap.
+CapabilityBoundingSet=CAP_NET_ADMIN CAP_SYS_ADMIN CAP_SYS_CHROOT CAP_SETUID CAP_SETGID
+MemoryDenyWriteExecute=yes
+NoNewPrivileges=yes
+PrivateDevices=yes
+# Breaks xapian-bridge, since it needs local networking
```
Comment (pwithnall) [inline data/eos-companion-app.service]: I think the wording here is unclear: “Breaks xapian-bridge” implies that `PrivateNetwork=no` breaks xapian-bridge, rather than `PrivateNetwork=yes` breaks it. I suggest rewording to “xapian-bridge needs local networking”.

Same for the other comment below.

File: data/eos-companion-app.service (hunk 4 above)
Comment (smspillaz) [reply data/eos-companion-app.service]: Oh weird, it is complaining about netlink now:

Jul 16 16:11:08 endless dbus-run-session[67331]: bwrap: loopback: Failed to create NETLINK_ROUTE socket: Address family not supported by protocol
